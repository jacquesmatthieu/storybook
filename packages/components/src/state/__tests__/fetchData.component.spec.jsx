import React from 'react';
import { shallow } from 'enzyme';

import FetchData, { withFetchData } from '../fetchData.component';

describe('FetchData', () => {
  let props;

  beforeEach(() => {
    props = {
      query: { foo: 'bar' },
    };
  });

  const getWrapper = () => shallow((
    <FetchData {...props}>
      {data => JSON.stringify(data)}
    </FetchData>
  ));

  it('should render loading state', () => {
    // When
    const wrapper = getWrapper();
    const subWrapper = wrapper.find('Query').renderProp('children', { loading: true });

    // Then
    expect(wrapper).toMatchSnapshot();
    expect(subWrapper).toMatchSnapshot();
  });

  it('should render error state', () => {
    // When
    const wrapper = getWrapper();
    const subWrapper = wrapper.find('Query').renderProp('children', { error: true });

    // Then
    expect(subWrapper).toMatchSnapshot();
  });

  it('should render loaded state', () => {
    // When
    const wrapper = getWrapper();
    const subWrapper = wrapper.find('Query').renderProp('children', { data: { hello: 'world' } });

    // Then
    expect(subWrapper).toMatchSnapshot();
  });

  it('should render wrapped component wioth HOC', () => {
    // Given
    const WrappedComponent = () => <div>Foo</div>;
    const Comp = withFetchData({ foo: 'bar' })(WrappedComponent);

    // When
    const wrapper = shallow(<Comp />);
    const subWrapper = wrapper.find('FetchData').renderProp('children', { hello: 'world' });

    // Then
    expect(wrapper).toMatchSnapshot();
    expect(subWrapper).toMatchSnapshot();
  });
});
