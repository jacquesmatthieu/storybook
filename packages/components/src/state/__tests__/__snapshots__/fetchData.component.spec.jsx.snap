// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FetchData should render error state 1`] = `
<AppError
  error={true}
  t={[MockFunction]}
/>
`;

exports[`FetchData should render loaded state 1`] = `"{\\"hello\\":\\"world\\"}"`;

exports[`FetchData should render loading state 1`] = `
<Query
  query={
    Object {
      "foo": "bar",
    }
  }
>
  <Component />
</Query>
`;

exports[`FetchData should render loading state 2`] = `
<WithCurrentApp(Loading)
  location={
    Object {
      "pathname": "foo",
    }
  }
  match={
    Object {
      "params": Object {},
    }
  }
/>
`;

exports[`FetchData should render wrapped component wioth HOC 1`] = `
<FetchData
  query={
    Object {
      "foo": "bar",
    }
  }
>
  <Component />
</FetchData>
`;

exports[`FetchData should render wrapped component wioth HOC 2`] = `
<WrappedComponent
  data={
    Object {
      "hello": "world",
    }
  }
/>
`;
